### ISSUE TITLE
Assignment to computed value misinterpreted in JDQL UPDATE when entity identification variable is ommitted

### ISSUE THREAD

**Issue by Riva-Tholoor-Philip** [eclipse-ee4j/eclipselink#2183]
**Describe the bug**

When the optional entity identification variable is omitted (for consistency with JDQL) of a JPQL query such as the following,

`UPDATE Coordinate SET x = :newX, y = y / :yDivisor WHERE id = :id`

EcilpseLink complains that "The left expression is not an arithmetic expression" instead of interpreting the assignment operation correctly.

```
Caused by: java.lang.IllegalArgumentException: An exception occurred while creating a query in EntityManager: 
Exception Description: Syntax error parsing [UPDATE Coordinate SET x = :newX, y = y / :yDivisor WHERE id = :id]. 
[37, 38] The left expression is not an arithmetic expression.
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.createQuery(EntityManagerImpl.java:1848)
	at io.openliberty.data.internal.persistence.RepositoryImpl.invoke(RepositoryImpl.java:1079)
	... 40 more
Caused by: Exception [EclipseLink-0] (Eclipse Persistence Services - 5.0.0-B02.v202404111748): org.eclipse.persistence.exceptions.JPQLException
Exception Description: Syntax error parsing [UPDATE Coordinate SET x = :newX, y = y / :yDivisor WHERE id = :id]. 
[37, 38] The left expression is not an arithmetic expression.
	at org.eclipse.persistence.internal.jpa.jpql.HermesParser.buildException(HermesParser.java:169)
	at org.eclipse.persistence.internal.jpa.jpql.HermesParser.validate(HermesParser.java:345)
	at org.eclipse.persistence.internal.jpa.jpql.HermesParser.populateQueryImp(HermesParser.java:292)
	at org.eclipse.persistence.internal.jpa.jpql.HermesParser.buildQuery(HermesParser.java:174)
	at org.eclipse.persistence.internal.jpa.EJBQueryImpl.buildEJBQLDatabaseQuery(EJBQueryImpl.java:144)
	at org.eclipse.persistence.internal.jpa.EJBQueryImpl.buildEJBQLDatabaseQuery(EJBQueryImpl.java:120)
	at org.eclipse.persistence.internal.jpa.EJBQueryImpl.<init>(EJBQueryImpl.java:107)
	at org.eclipse.persistence.internal.jpa.EJBQueryImpl.<init>(EJBQueryImpl.java:91)
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.createQuery(EntityManagerImpl.java:1846)
```

**Comment by ajaypaul-ibm** [comment]
We were able to recreate this issue using a simple JPA application. Below is the Coordinate entity we created by referencing the entity present in Jakarta Data.
```
@Entity
public class Coordinate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private double x;

    private float y;
}
```
Here is the code where we executed the JPQL query:

```
EntityTransaction transaction = entityManager.getTransaction();
transaction.begin();

entityManager.createQuery("UPDATE Coordinate SET x = :newX, y = y / :yDivisor WHERE id = :id")
        .setParameter("newX", newX)
        .setParameter("yDivisor", yDivisor)
        .setParameter("id", id)
        .executeUpdate();

transaction.commit();
```
This resulted in the following exception stack, which is the same as described in the issue:

```
Exception in thread "main" java.lang.IllegalArgumentException: An exception occurred while creating a query in EntityManager: 
Exception Description: Syntax error parsing [UPDATE Coordinate SET x = :newX, y = y / :yDivisor WHERE id = :id]. 
[37, 38] The left expression is not an arithmetic expression.
    at org.eclipse.persistence.internal.jpa.EntityManagerImpl.createQuery(EntityManagerImpl.java:1751)
    at com.example.repositories.CoordinateRepository.updateCoordinate(CoordinateRepository.java:18)
    at org.example.Main.main(Main.java:34)
Caused by: Exception [EclipseLink-0] (Eclipse Persistence Services - 2.7.9.v20210604-2c549e2208): org.eclipse.persistence.exceptions.JPQLException
Exception Description: Syntax error parsing [UPDATE Coordinate SET x = :newX, y = y / :yDivisor WHERE id = :id]. 
[37, 38] The left expression is not an arithmetic expression.
    at org.eclipse.persistence.internal.jpa.jpql.HermesParser.buildException(HermesParser.java:157)
    at org.eclipse.persistence.internal.jpa.jpql.HermesParser.validate(HermesParser.java:336)
    at org.eclipse.persistence.internal.jpa.jpql.HermesParser.populateQueryImp(HermesParser.java:280)
    at org.eclipse.persistence.internal.jpa.jpql.HermesParser.buildQuery(HermesParser.java:165)
    at org.eclipse.persistence.internal.jpa.EJBQueryImpl.buildEJBQLDatabaseQuery(EJBQueryImpl.java:142)
    at org.eclipse.persistence.internal.jpa.EJBQueryImpl.buildEJBQLDatabaseQuery(EJBQueryImpl.java:118)
    at org.eclipse.persistence.internal.jpa.EJBQueryImpl.<init>(EJBQueryImpl.java:104)
    at org.eclipse.persistence.internal.jpa.EJBQueryImpl.<init>(EJBQueryImpl.java:88)
    at org.eclipse.persistence.internal.jpa.EntityManagerImpl.createQuery(EntityManagerImpl.java:1749)
    ... 2 more
```

**Comment by ajaypaul-ibm** [comment]
FYI , I'm currently looking into this issue.

**Comment by OndroMih** [comment]
I believe this case is already fixed in the master branch with https://github.com/eclipse-ee4j/eclipselink/pull/2218.

**Comment by Riva-Tholoor-Philip** [comment]
> I believe this case is already fixed in the master branch with #2218.

@OndroMih  - so the PR [#2218](https://github.com/eclipse-ee4j/eclipselink/pull/2218) will fix two issues [2183](https://github.com/eclipse-ee4j/eclipselink/issues/2183) and [2187](https://github.com/eclipse-ee4j/eclipselink/issues/2187)

**Comment by OndroMih** [comment]
Yes. I fixed some issues with implicit alias too, because I hit them with a test query that I used to reproduce the main issue with “lenght”.
